processor/src/chiplets/ace/mod.rs

```diff
@@ -269,3 +277,83 @@ impl AceHints {
             .sum()
     }
 }
+
+/// Evaluates an arithmetic circuit at `(ctx, clk)` given a pointer `ptr` to its description,
+/// the number of variables/inputs to the circuit and the number of evaluation gates.
+///
+/// The description of the circuit is divided into two portions:
+///
+/// 1. `READ` made up of the inputs to the circuit followed by the constants of the circuit, both of
+///    which are quadratic extension field elements,
+/// 2. `EVAL` made up of the base field elements encoding each evaluation gate of the circuit. Each
+///    gate is encoded as `[ id_l (30 bits) || id_r (30 bits) || op (2 bits) ]`, where `id_l` is the
+///    identifier of the left input wire, `id_r` is the identifier of the right input wire and `op`
+///    is the operation executed by the gate, namely `op ∈ {0, 1, 2}` where `0` denotes a `SUB`, `1`
+///    a `MUL` and `2` an `ADD`.
+#[allow(dead_code)]
+pub fn eval_circuit(
```

Comment:
Why did we need to add this function in this PR? Also, it seems like it is used in tests - shouldn't it be marked as `#[cfg(test)]`?